main_objs.py

```diff
@@ -448,9 +448,30 @@ def getHarmonic(self, kind='q', directed=True, compound=True):
             self.analyses[key] = df
         return self.analyses[key]
 
-    def _ngramHelper(col, n, exclude, cell_type):
+    def _ngrams_offsets_helper(col, n, offsets):
+        if offsets == 'start':
+            chunks = [col.shift(-i) for i in range(n)]
+        elif offsets == 'end':
+            chunks = [col.shift(i) for i in range(n - 1, -1, -1)]
+        else:
+            raise Exception("Please specify offsets options!")
```

Comment:
I don't think we need validation checks for the arguments of a private method, so we don't need this else clause. It would be more helpful to add a docstring instead.
More importantly, this is a solid use of .shift(). Good job.